### ISSUE TITLE
GET endpoints stopped working after updating to 7.3.0

### ISSUE THREAD

**Issue by AcidSlide** [nestjs/nest#5011]
## Regression
All "GET" endpoints stopped working after updating @nestjs/core and @nestjs/common from 7.2.0 to 7.3.0.

NOTE: We are also using platform-fastify as the wrapper framework.

## Potential Commit/PR that introduced the regression**
Updating to 7.3.0 from 7.2.0

## Describe the regression
After updating NestJS modules @nestjs/core and @nestjs/common to 7.3.0, all of our GET endpoints are returning "null".  We have retested with version 7.2.0 of the packages and it's working, it only stopped working after updating NestJS.

## Input Code
sorry, no time to create a sample repo.

```ts
  @Get()
  getHello(): string {
    return 'This is a test Hello World!';
  }
```

## Expected behavior/code
GET endpoints should work

## Environment
<pre><code>
 _   _             _      ___  _____  _____  _     _____
| \ | |           | |    |_  |/  ___|/  __ \| |   |_   _|
|  \| |  ___  ___ | |_     | |\ `--. | /  \/| |     | |
| . ` | / _ \/ __|| __|    | | `--. \| |    | |     | |
| |\  ||  __/\__ \| |_ /\__/ //\__/ /| \__/\| |_____| |_
\_| \_/ \___||___/ \__|\____/ \____/  \____/\_____/\___/

[System Information]
OS Version     : macOS Catalina
NodeJS Version : v10.20.1
NPM Version    : 6.14.5

[Nest CLI]
Nest CLI Version : 7.4.1

[Nest Platform Information]
platform-express version : 7.2.0
platform-fastify version : 7.2.0
passport version         : 7.0.0
swagger version          : 4.5.11
typeorm version          : 7.1.0
common version           : 7.3.0
core version             : 7.3.0
jwt version              : 7.0.0
</code></pre>

**Comment by kamilmysliwiec** [comment]
I've tested with samples (/sample/ dir) as well as double-checked our integration tests - everything seems to work fine. I can't really reproduce your issue. Can you share a minimal repository?

**Comment by AcidSlide** [comment]
we'll try to create a minimal repository based on the bare minimum of our system

**Comment by AcidSlide** [comment]
@kamilmysliwiec here is the minimal recreation of the issue.. GET endpoints returns "null"

https://github.com/AcidSlide/nestjs-get-issue-5011

**Comment by kamilmysliwiec** [comment]
Can you please create a minimal repository without additional validators, TypeORM, Mailer, external dependencies etc?

Also, please test if this issue still appears in `7.3.1` (there was a small fix to the `CacheInterceptor` that you use in your app).

**Comment by AcidSlide** [comment]
> Can you please create a minimal repository without additional validators, TypeORM, Mailer, external dependencies etc?
> 
> Also, please test if this issue still appears in `7.3.1` (there was a small fix to the `CacheInterceptor` that you use in your app).

Ok we will try first updating then test, then test without the external dependencies mentioned

**Comment by AcidSlide** [comment]
@kamilmysliwiec without removing the additional dependencies, everything worked with 7.3.1.. must be related to the `CacheInterceptor` you mentioned.

**Comment by AcidSlide** [comment]
We'll try the update to our full build if everything is working again

**Comment by AcidSlide** [comment]
Based on our test with our full system, this looks like it has been fixed with 7.3.1. Closing it now.
